### PR TITLE
docs: add data collection (telemetry) notice to readme

### LINKED ISSUE #2673: Add data collection notice to readme
Collecting telemetry requires a notice to the community in the readme of the repo.

### PR DESCRIPTION
**Pull request by devigned**
**Reason for Change**:

If a Microsoft open source project or release has source code that enables Microsoft to collect data/telemetry, you must place in the README a notice.

This PR satisfies our community and legal requirement.

**Issue Fixed**:

fixes #2673 

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [x] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by craiglpeters** [comment]
Two questions:
1. Is it clear to a user what data is collected, and where it is sent?
1. Is it clear to a user how to turn this off?

File: README.md [hunk 1]
```diff
@@ -38,3 +38,6 @@ This project has adopted the [Microsoft Open Source Code of Conduct](https://ope
 [docs]: docs/README.md
 [FAQ]: docs/faq.md
 [tutorials]: docs/tutorials/README.md
+
+## Data Collection
+The software may collect information about you and your use of the software and send it to Microsoft. Microsoft may use this information to provide services and improve our products and services. You may turn off the telemetry as described in the repository. There are also some features in the software that may enable you and Microsoft to collect data from users of your applications. If you use these features, you must comply with applicable law, including providing appropriate notices to users of your applications together with a copy of Microsoft's privacy statement. Our privacy statement is located at https://go.microsoft.com/fwlink/?LinkID=824704. You can learn more about data collection and use in the help documentation and our privacy statement. Your use of the software operates as your consent to these practices.
```
Comment (CecileRobertMichon) [inline README.md]:
```suggestion
The software may collect information about you and your use of the software and send it to Microsoft. Microsoft may use this information to provide services and improve our products and services. You may turn off the telemetry as described in the repository. There are also some features in the software that may enable you and Microsoft to collect data from users of your applications. If you use these features, you must comply with applicable law, including providing appropriate notices to users of your applications together with a copy of Microsoft's privacy statement. You can learn more about data collection and use in the help documentation and in our [privacy statement](https://go.microsoft.com/fwlink/?LinkID=824704). Your use of the software operates as your consent to these practices.
```

Comment (devigned) [reply README.md]: I would, but I literally copy and pasted this from our internal OSS guidance site. I'm hesitant to change any of the wording. Thoughts?

**Comment by devigned** [comment]
@craiglpeters great questions!

- Being clear about turning off telemetry
  - I added a bunch of details to the ./docs/topics/telemetry.md file in [this PR](https://github.com/Azure/aks-engine/pull/2606)
- What data is being collected
  - What do you think about updating ./docs/topics/telemetry.md to also contain that information?
  - There's still a little work here to finish up the instance metadata collection, but after that, I can show some samples of data collected with all the fields and representative values.

File: README.md (hunk 1 above)
Comment (palmerabollo) [inline README.md]: I'd link "as described in the repository" to the specific markdown file describing how to configure it. For example, to https://github.com/Azure/aks-engine/blob/master/docs/topics/telemetry.md#configuration.

**Comment by craiglpeters** [comment]
> * Being clear about turning off telemetry
>   
>   * I added a bunch of details to the ./docs/topics/telemetry.md file in [this PR](https://github.com/Azure/aks-engine/pull/2606)
LGTM with a single comment there

> * What data is being collected
>   
>   * What do you think about updating ./docs/topics/telemetry.md to also contain that information?
That makes sense to me
>   * There's still a little work here to finish up the instance metadata collection, but after that, I can show some samples of data collected with all the fields and representative values.

**Pull request merged**